### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by gfzabarino**
Fixed the following bugs:
- Input toolbar was getting hidden when pushing a view controller from a JSQMessagesViewController, but when popping back, the keyboard was visible and the input toolbar was not.
- Panning handling was using wrong coordinates for calculation of where to start panning.

Hey, what do you think about these changes?

### CODE REVIEW

File: JSQMessagesViewController/Controllers/JSQMessagesKeyboardController.m [hunk 1]
```diff
@@ -312,7 +311,7 @@ - (void)jsq_removeKeyboardFrameObserver
 
 - (void)jsq_handlePanGestureRecognizer:(UIPanGestureRecognizer *)pan
 {
-    CGPoint touch = [pan locationInView:self.contextView];
+    CGPoint touch = [pan locationInView:self.contextView.window];
```
Comment (jessesquires) [inline JSQMessagesViewController/Controllers/JSQMessagesKeyboardController.m]: looks like this fixes #1161

**Comment by jessesquires** [comment]
Thanks @gfzabarino :smile: 

Looks like this will definitely cause some regressions/bugs.

However, I will include the pan gesture fix! :smile:

**Comment by gfzabarino** [comment]
No problem @jessesquires. Are you sure there will be bugs / regressions because of this? I could agree regarding removing the dispatch_async call, but other than that I don't think the proposed changes will do anything wrong. The issue I'm describing is very annoying actually.

**Comment by jessesquires** [comment]
There's a lot of history (legacy) around why the begin/end listening calls are where they are. I'm hesitant to touch that.

I'm planning to revisit this for the 8.0 release.

**Pull request closed without merging**